src/ksym.rs

```diff
@@ -91,29 +91,69 @@ impl KSymResolver {
         }
     }
 
-    pub fn find_address_ksym(&self, addr: u64) -> Option<&Ksym> {
+    pub fn find_address_ksym(&self, addr: u64) -> Vec<&Ksym> {
```

Comment:
Would it be possible that we use https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search_by instead of rolling our own binary search here, twice?

I don't really understand what `dropped_l`'s purpose is and what high level problem we solve with those two loops, to be honest. Can you explain?

Reply:
According to the document,  it says
```
If the value is found then [Result::Ok](https://doc.rust-lang.org/std/result/enum.Result.html#variant.Ok) is returned, 
containing the index of the matching element. If there are multiple matches, then any one of the matches could be 
returned.
```

The problem here is the potential of having multiple symbols at the same address.

Reply:
What I am saying is that it could be used to find the starting position, no? I.e., the one where 
> all symbols with addresses that are less than or equal to addr will be at the left side of `l` except the last one. All symbols with addresses that are bigger than addr will be at the right side of `l`

Reply:
If you are saying to replace the first while-loop with the built-in binary search, yes.  However, we will miss dropped_l.  That means we need to search in the range (0..l] instead of the range (dropped_l..l].

Reply:
The methods accepts a `FnMut`, meaning you should be able to write to the environment from it. I.e., you can close over the `dropped_l` variable and modify it from the closure body, while abstracting away the finer binary search details.

Reply:
But...now that it has become clearer what the code is trying to do, the method does not support looking at more than one element at a time, as in Andrii's condition: `ksym[l].addr <= addr && addr < ksym[l + 1].addr`. So if that's required, we have to hand roll. If we wanted to use it, we'd have to scan both directions for additional matches. In that case it's more of a trade-off. I am fine either way.